### DISCUSSION TITLE
It is possible to add the dictionary arguments in the linkout function?

### DISCUSSION THREAD

**Discussion by Chi-HsienChang** [GMOD/jbrowse-components#3640] [Q&A] [answered]
Hi JBrowse2 developers,

I want to ask:

It is possible to add the dictionary arguments in the linkout function?

Here's I currently try do it as follows:

![image](https://user-images.githubusercontent.com/114167559/231529676-5fa65a79-240d-4f45-8984-39a8a7931d72.png)
![image](https://user-images.githubusercontent.com/114167559/231530139-560f9d0a-7426-49e5-9f93-0237a3b35035.png)

After that, the Feature details turn to empty:
![image](https://user-images.githubusercontent.com/114167559/231530392-e58b33f5-a146-46d0-984e-e63cf2185949.png)

I try the grammer of TypeScript is correct:

![image](https://user-images.githubusercontent.com/114167559/231530660-e83c71ae-f880-40e9-a4ac-12a17ed13103.png)

This is the code of the plugin:
https://github.com/Chi-HsienChang/jbrowse-plugin-linkout

Thank you so much!

**Comment by cmdcolin** [comment]
as far as I can tell, passing an object just like you do works for me. I would again check the javascript console to see if there is any bug, but it look ok to me. You mentioned the blank screen also yesterday, I am not sure what causes that but if you can identify make a small reproducible case, I can investigate. https://github.com/GMOD/jbrowse-components/discussions/3635#discussioncomment-5583734


that said, I will note, stylistically, the way you are looping through the object with a double nested loop looks odd to me. i recommend pasting raw source code and not a screenshot as it's hard to copy and paste otherwise, and (personal preference :)) recommend running through 
 https://prettier.io/

**Reply by cmdcolin** [reply]
I see you pasted code here also so that is helpful https://github.com/Chi-HsienChang/jbrowse-plugin-linkout

here is a refactoring i tried

```typescript
pluginManager.jexl.addFunction('linkout', (dict, feature) => {
        if (!feature.dbxref) {
          return ''
        }
        const dbxrefs = Array.isArray(feature.dbxref)
          ? feature.dbxref
          : [feature.dbxref]

        return dbxrefs.map(dbxref => {
          const [root, content] = dbxref.split(':')
          const link = dict[root]
          return link ? `<a href=${link}${content}>${dbxref}</a>` : dbxref
        })
      })
```

then in config

```
"formatDetails": {
        "feature": "jexl:{dbxref:linkout({Genbank:'https://www.ncbi.nlm.nih.gov/nuccore/'},feature)}",
        "subfeatures": "jexl:{dbxref:linkout({Genbank:'https://www.ncbi.nlm.nih.gov/nuccore/'},feature)}"
      },
```

**Reply by cmdcolin** [reply, marked as answer]
I also tried making a better error handler in this PR so that in future versions you won't get a blank screen on error but instead get an error message https://github.com/GMOD/jbrowse-components/pull/3642/files

**Comment by Chi-HsienChang** [comment]
Hi @cmdcolin, 

I really appreciate you so much!!!!!!!

It works for me now!
